### PR TITLE
Add Spark connector settings

### LINKED ISSUE #3454: [DOC] Spark Connector
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
Introduce the ability to reach data at rest in an object store (e.g., S3 or data lake) via Spark (Spark to be supplied by the user). Our first release will focus on data stored in Parquet format. Users will be able to query this data as though it is regular OpenSearch data and visualize it alongside their OpenSearch data in OpenSearch Dashboards.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/opensearch-spark/issues/4
https://github.com/opensearch-project/sql/issues/1746

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #3454 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/sql/settings.md [hunk 1]
```diff
@@ -78,3 +78,37 @@ Setting | Default | Description
 `plugins.sql.cursor.keep_alive` | 1 minute | Configures how long the cursor context is kept open. Cursor contexts are resource-intensive, so we recommend a low value.
 `plugins.query.memory_limit` | 85% | Configures the heap memory usage limit for the circuit breaker of the query engine.
 `plugins.query.size_limit` | 200 | Sets the default size of index that the query engine fetches from OpenSearch.
+
+## Spark connector settings
+
+The SQL plugin supports [Apache Spark](https://spark.apache.org/) as a data source. This allows you to run SQL queries in Spark and seamlessly write the resulting data back to OpenSearch. To use Spark, enable the following settings to add Spark as a data source and enable the correct permissions.
```
Comment (brijos) [inline _search-plugins/sql/settings.md:84]: The SQL plugin supports Apache Spark as an augmented compute source. Data sources defined in Apache Spark as tables will be consumable from OpenSearch. OpenSearch users will be able to execute SQL queries against external sources from within OpenSearch Dashboards Discover and Observability Logs. To get started, enable the following settings to add Spark as a data source and enable the correct permissions.

**Review by brijos — CHANGES_REQUESTED** [review]

File: _search-plugins/sql/settings.md (hunk 1 above)
Comment (Naarcha-AWS) [reply _search-plugins/sql/settings.md:84]: Switched it to this: 

---

The SQL plugin supports [Apache Spark](https://spark.apache.org/) as an augmented compute source. When data sources are defined as tables in Apache Spark, OpenSearch can consume those tables. This allows you to run SQL queries against external sources inside OpenSearch Dashboard's [Discover]({{site.url}}{{site.baseurl}}/dashboards/discover/index-discover/) and observability logs. 

To get started, enable the following settings to add Spark as a data source and enable the correct permissions.

Comment (brijos) [reply _search-plugins/sql/settings.md]: That works.

File: _search-plugins/sql/settings.md [hunk 2]
```diff
@@ -78,3 +78,39 @@ Setting | Default | Description
...
+`spark.datasource.flint.scheme` | The data scheme used in your Spark queries. Valid values are `http` and `https`.
+`spark.datasource.flint.auth` | The authorization required to access the Spark data source. Valid values are `false` and `sigv4`.
+`spark.datasource.flint.region` | The region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.
+`spark.datasource.flint.write.id_name` | The index name in which the Spark connector writes.
+`spark.datasource.flint.ignore.id_column` | Excludes the `id` column when exporting data in a query. Default is `true`.
+`spark.datasource.flint.write.batch_size` | Sets the batch size when writing to a Spark-connected index. Default is `1000`.
+`spark.datasource.flint.write.refresh_policy` | Sets the refresh policy for the Spark connection upon failure for the connector to write data to OpenSearch. Either no refresh (`false`), an immediate refresh (`true`), or a set time to wait `wait_for: X`. Default value is `false`.
+`spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries ran with Spark. Default is `100`.
```
Comment (cwillum) [inline _search-plugins/sql/settings.md:103]:
```suggestion
`spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries run with Spark. Default is `100`.
```

**Review by cwillum — APPROVED** [review]
LGTM

File: _search-plugins/sql/settings.md [hunk 3]
```diff
@@ -78,3 +78,39 @@ Setting | Default | Description
...
+:--- | :---
+`spark.uri` | The identifier for your Spark data source.
+`spark.auth.type` | The authorization type used to authenticate into Spark.
+`spark.auth.username` | The username for your Spark data source.
+`spark.auth.password` | The password for your Spark data source.
+`spark.datasource.flint.host` | The host of the Spark data source. Default is `localhost`.
+`spark.datasource.flint.port` | The port number for Spark. Default is `9200`.
+`spark.datasource.flint.scheme` | The data scheme used in your Spark queries. Valid values are `http` and `https`.
+`spark.datasource.flint.auth` | The authorization required to access the Spark data source. Valid values are `false` and `sigv4`.
+`spark.datasource.flint.region` | The region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.
```
Comment (natebower) [inline _search-plugins/sql/settings.md:98]:
```suggestion
`spark.datasource.flint.region` | The AWS Region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.
```

Comment (natebower) [inline _search-plugins/sql/settings.md:98]: When referring to an AWS Region, always use "AWS Region" on first appearance. Capitalized "Region" may be used for subsequent appearances.

File: _search-plugins/sql/settings.md [hunk 4]
```diff
@@ -78,3 +78,39 @@ Setting | Default | Description
...
+`spark.datasource.flint.write.id_name` | The index name in which the Spark connector writes.
+`spark.datasource.flint.ignore.id_column` | Excludes the `id` column when exporting data in a query. Default is `true`.
+`spark.datasource.flint.write.batch_size` | Sets the batch size when writing to a Spark-connected index. Default is `1000`.
+`spark.datasource.flint.write.refresh_policy` | Sets the refresh policy for the Spark connection upon failure for the connector to write data to OpenSearch. Either no refresh (`false`), an immediate refresh (`true`), or a set time to wait `wait_for: X`. Default value is `false`.
```
Comment (natebower) [inline _search-plugins/sql/settings.md:102]:
```suggestion
`spark.datasource.flint.write.refresh_policy` | Sets the refresh policy for the Spark connection upon failure for the connector to write data to OpenSearch. Either no refresh (`false`), an immediate refresh (`true`), or a set time to wait, `wait_for: X`. Default value is `false`.
```

File: _search-plugins/sql/settings.md [hunk 5]
```diff
@@ -78,3 +78,39 @@ Setting | Default | Description
...
+`spark.datasource.flint.read.scroll_size` | Sets the amount of results returned by queries run with Spark. Default is `100`.
```
Comment (natebower) [inline _search-plugins/sql/settings.md:103]:
```suggestion
`spark.datasource.flint.read.scroll_size` | Sets the number of results returned by queries run using Spark. Default is `100`.
```

File: _search-plugins/sql/settings.md [hunk 6]
```diff
@@ -78,3 +78,39 @@ Setting | Default | Description
...
+`spark.uri` | The identifier for your Spark data source.
+`spark.auth.type` | The authorization type used to authenticate into Spark.
+`spark.auth.username` | The username for your Spark data source.
+`spark.auth.password` | The password for your Spark data source.
+`spark.datasource.flint.host` | The host of the Spark data source. Default is `localhost`.
+`spark.datasource.flint.port` | The port number for Spark. Default is `9200`.
+`spark.datasource.flint.scheme` | The data scheme used in your Spark queries. Valid values are `http` and `https`.
+`spark.datasource.flint.auth` | The authorization required to access the Spark data source. Valid values are `false` and `sigv4`.
+`spark.datasource.flint.region` | The region in which your OpenSearch cluster is located. Only use when `auth` is set to `sigv4`. Default value is `us-west-2``.
+`spark.datasource.flint.write.id_name` | The index name in which the Spark connector writes.
```
Comment (natebower) [inline _search-plugins/sql/settings.md:99]:
```suggestion
`spark.datasource.flint.write.id_name` | The name of the index to which the Spark connector writes.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Just a few minor changes, as discussed. Thanks!

**Pull request merged**